server/service/service_osquery.go

```diff
@@ -447,22 +445,19 @@ func (svc *Service) detailQueriesForHost(ctx context.Context, host *fleet.Host)
 	return queries, nil
 }
 
-func (svc *Service) shouldUpdate(lastUpdated time.Time, interval time.Duration) bool {
+func (svc *Service) shouldUpdate(lastUpdated time.Time, interval time.Duration, hostID uint) bool {
 	var jitter time.Duration
 	if svc.config.Osquery.MaxJitterPercent > 0 {
-		maxJitter := time.Duration(svc.config.Osquery.MaxJitterPercent) * interval / time.Duration(100.0)
-		randDuration, err := rand.Int(rand.Reader, big.NewInt(int64(maxJitter)))
-		if err == nil {
-			jitter = time.Duration(randDuration.Int64())
-		}
+		maxJitter := int64(svc.config.Osquery.MaxJitterPercent) * int64(interval) / 100.0
+		jitter = time.Duration((int64(hostID) + svc.getJitterSeed()) % maxJitter)
```

Comment:
I shouldn't comment on such math before coffee, double- and triple-check what follows, but I think this will generate milli-micro-nano jitters most of the time? I.e. `interval` is a number in nanoseconds, so `maxJitter` is some fraction of this (and typically >= 10%, so still a large number representing nanoseconds), wouldn't a large portion of the resulting `jitter` be < 1s (and thus make very little difference on randomizing the requests)?

Reply:
Based on @lucasmrod tests: https://github.com/fleetdm/fleet/pull/3716#discussion_r786184778, this works.

Reply:
Ah yes ideed, running it in a loop (https://go.dev/play/p/mHbcH2EKBsz), it's basically 1 or 2 per 1000 that is below a second, seems fine.